src/test/java/io/neonbee/config/NeonBeeConfigTest.java

```diff
@@ -48,16 +64,65 @@ protected WorkingDirectoryBuilder provideWorkingDirectoryBuilder(TestInfo testIn
+    void testLoadingMeterRegistry() {
+        NeonBeeConfig config = new NeonBeeConfig();
+        config.setMicrometerRegistries(List.of(TestMicrometerRegistryLoaderImpl.class.getName()));
+        MicrometerMetricsOptions micrometerOptions = NeonBeeConfig.loadMicrometerOptions(config);
+        CompositeMeterRegistry registry = (CompositeMeterRegistry) micrometerOptions.getMicrometerRegistry();
+        Set<MeterRegistry> registries = registry.getRegistries();
+        assertThat(registries).hasSize(2);
+        assertThat(registries.stream().anyMatch(PrometheusMeterRegistry.class::isInstance)).isTrue();
+        assertThat(registries.stream().anyMatch(LoggingMeterRegistry.class::isInstance)).isTrue();
+    }
+
+    static Stream<Arguments> testNotImplementingMicrometerRegistryLoaderArguments() {
+        return Stream.of(
+                Arguments.of("java.lang.String",
+                        "java.lang.String must implement io.neonbee.config.metrics.MicrometerRegistryLoader",
+                        IllegalArgumentException.class),
+                Arguments.of("doesn't exist", "Could not load custom micrometer registry", IllegalStateException.class),
+                Arguments.of("io.neonbee.config.NeonBeeConfigTest$TestFaultyMicrometerRegistryLoaderImpl",
+                        "Could not load custom micrometer registry", IllegalStateException.class));
+    }
+
+    @ParameterizedTest(name = "{index}: {0} expected exception message: {1}")
+    @MethodSource("testNotImplementingMicrometerRegistryLoaderArguments")
+    @Timeout(value = 2, timeUnit = TimeUnit.SECONDS)
+    @DisplayName("Test MicrometerRegistryLoader with incorrect configuration")
+    void testNotImplementingMicrometerRegistryLoader(String className, String exceptionMessage,
+            Class expectedException) {
+        NeonBeeConfig config = new NeonBeeConfig();
+        config.setMicrometerRegistries(List.of(className));
+        Throwable throwable = assertThrows(expectedException, () -> NeonBeeConfig.loadMicrometerOptions(config));
+        assertThat(throwable).hasMessageThat().isEqualTo(exceptionMessage);
+    }
+
     @Test
     @Timeout(value = 2, timeUnit = TimeUnit.SECONDS)
     @DisplayName("should load NeonBeeConfig correctly from working dir")
+    @SuppressWarnings({ "deprecation", "removal" })
     void testLoad(Vertx vertx, VertxTestContext testContext) {
         NeonBeeConfig.load(vertx).onComplete(testContext.succeeding(nbc -> {
             testContext.verify(() -> isEqualToDummyConfig(nbc));
             testContext.completeNow();
         }));
     }
 
+    @Test
+    @Timeout(value = 2, timeUnit = TimeUnit.SECONDS)
+    @DisplayName("should load NeonBeeConfig correctly from working dir")
+    @SuppressWarnings("SuppressWarningsDeprecated")
+    void testLoadNeonBeeConfigBlocking() throws Exception {
+        NeonBeeConfig config = NeonBeeConfig.load(Path.of("src", "test", "resources", "io", "neonbee", "config"));
```

Comment:
TEST_RESOURCES.resolve("io/neonbee/config")

or

Path.of("src/test/resources/io/neonbee/config")

Reply:
makes sense to use the TEST_RESOURCES resolver